test/smoke/testApps/AutoPerfCounters/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/PerfCountersDataTest.java

```diff
@@ -0,0 +1,163 @@
+                    return false;
+                return "RequestData".equals(input.getData().getBaseType());
+            }
+        }, 10, TimeUnit.SECONDS); // this one just returns a single envelope.
+        // at this point, the item has already arrived and this just returns immediately
+        assertNotNull(requestItem);
+        System.out.println("Requests are good: " + (System.currentTimeMillis() - start));
+
+        int heapMemoryUsed = 0;
+        int suspected = 0;
+        for (int i = metricItems.size(); i > 2; i = i - 1) {
+            DataPoint dp1 = getMetricDataDetails(i - 1);
+            switch (dp1.getName()) {
+            case "Suspected Deadlocked Threads":
+                assertEquals(DataPointType.Measurement, dp1.getKind());
+                assertTrue(dp1.getValue() == 0);
+                suspected++;
+                break;
+            case "Heap Memory Used (MB)":
+                assertEquals(DataPointType.Measurement, dp1.getKind());
+                assertTrue(dp1.getValue() > 0);
+                heapMemoryUsed++;
+                break;
+            default:
+                break;
+            }
+        }
+        assertEquals(1, heapMemoryUsed);
+        assertEquals(1, suspected);
+
+        int processor = 0, processPrivate = 0, processIO = 0, processTime = 0, memory = 0;
+        for (int i = performanceItems.size(); i > 3; i = i - 1) {
+            PerformanceCounterData perfd1 = getTelemetryDataForType(i - 1, "PerformanceCounterData");
+            switch (perfd1.getCategoryName()) {
+            case "Processor":
+                assertEquals("% Processor Time", perfd1.getCounterName());
+                assertEquals("_Total", perfd1.getInstanceName());
+                processor++;
+                break;
+            case "Memory":
+                assertEquals("Available Bytes", perfd1.getCounterName());
+                assertTrue(perfd1.getValue() != 0);
+                memory++;
+            case "Process":
+                assertTrue(perfd1.getValue() != 0);
+                switch (perfd1.getCounterName()) {
+                case "Private Bytes":
+                    processPrivate++;
+                    break;
+                case "IO Data Bytes/sec":
+                    processIO++;
+                    break;
+                case "% Processor Time":
+                    processTime++;
+                    break;
+                default:
+                    break;
+                }
+            default:
+                break;
+            }
+        }
+
+        assertEquals(1, processor);
+        assertEquals(1, processPrivate);
+        assertEquals(1, processIO);
+        assertEquals(1, processTime);
+        assertEquals(1, memory);
+    }
+
+    private DataPoint getMetricDataDetails(int index) {
+        MetricData md = getTelemetryDataForType(index, "MetricData");
+        List<DataPoint> metrics = md.getMetrics();
+        assertEquals(1, metrics.size());
```

Comment:
@Terryxxx , continuing from my other comment. Since this whole test is about perfcounters, all the metrics should be tagged with `"CustomPerfCounter"` set to `"true"`. You can probably add the assert statement here.